src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -0,0 +1,400 @@
+
+//     ModelDesc desc;
+//     desc.type = config_data["model_type"].get<std::string>();
+//     // NB: In case _name_or_path field isn't presented in config.json
+//     if (config_data.contains("_name_or_path")) {
+//         desc.name_or_path = config_data["_name_or_path"].get<std::string>();
+//     }
+//     desc.num_key_value_heads = config_data["num_key_value_heads"].get<int>();
+//     return desc;
+// }
+
+bool is_cw_compressed(const std::shared_ptr<ov::Model>& model) {
+    std::vector<std::string> rt_info_path = {"nncf", "weight_compression", "group_size"};
+    if (!model->has_rt_info(rt_info_path)) {
+        // NB: Model isn't compressed by NNCF - skip
+        return false;
+    }
+    auto group_size = model->get_rt_info<int>(rt_info_path);
+    if (group_size == -1) {
+        // NB: Enable DQ for CW quantized models
+        return true;
+    }
+    return false;
+}
+
+uint32_t align_to(uint32_t value, uint32_t alignment) {
+    return (value + alignment - 1) & ~(alignment - 1);
+}
+
+struct NPUDesc {
+    std::string arch;
+    int64_t max_tiles;
+};
+
+std::optional<NPUDesc> extract_npu_descriptor(const std::shared_ptr<const ov::IPlugin>& plugin) {
+    const ov::Any arch = plugin->get_property(ov::device::architecture.name(), ov::AnyMap{});
+    const ov::Any max_tiles = plugin->get_property(ov::intel_npu::max_tiles.name(), ov::AnyMap{});
+    return std::make_optional(NPUDesc{arch.as<std::string>(), max_tiles.as<int64_t>()});
+}
+
+std::optional<ov::Any> pop_option(ov::AnyMap& config, const std::string& option_name) {
+    if (auto it = config.find(option_name); it != config.end()) {
+        std::optional<ov::Any> found = std::make_optional(it->second);
+        config.erase(it);
+        return found;
+    }
+    return std::nullopt;
+}
+
+template <typename T>
+std::optional<T> get_option(ov::AnyMap& config, const std::string& option_name) {
+    if (auto it = config.find(option_name); it != config.end()) {
+        return std::make_optional(it->second.as<T>());
+    }
+    return std::nullopt;
+}
+
+template <typename T>
+T pop_or_default(ov::AnyMap& config, const std::string& key, const T& default_value) {
+    auto anyopt = pop_option(config, key);
+    if (anyopt.has_value()) {
+        return anyopt.value().as<T>();
+    }
+    return default_value;
+}
+
+enum class GenerateHint {
+    FAST_COMPILE,
+    BEST_PERF
+};
+
+GenerateHint str_to_hint(const std::string& str) {
+    if (str == "FAST_COMPILE") {
+        return GenerateHint::FAST_COMPILE;
+    }
+    if (str == "BEST_PERF") {
+        return GenerateHint::BEST_PERF;
+    }
+    OPENVINO_THROW("Unsupported \"GENERATE_HINT\" provided: " +
+                   str + ". Please select either \"FAST_COMPILE\" or \"BEST_PERF\".");
+}
+
+ov::AnyMap get_baseline_common_config() {
+    ov::AnyMap config = {
+        { "NPU_COMPILATION_MODE_PARAMS", "compute-layers-with-higher-precision=Sqrt,Power,ReduceMean,Add_RMSNorm" },
+        { "NPUW_DEVICES", "NPU,CPU" },
```

Comment:
CPU must be deleted here. Where does this map come from?

Reply:
From GenAI pipeline

Reply:
@TolyaTalamanov didn't you delete CPU from the devices?

Reply:
@AsyaPronina DELETE CPU

Reply:
Removed!